pkg/controller/endpointslice/utils.go

```diff
@@ -346,3 +350,62 @@ func (sl endpointSliceEndpointLen) Swap(i, j int) { sl[i], sl[j] = sl[j], sl[i]
 func (sl endpointSliceEndpointLen) Less(i, j int) bool {
 	return len(sl[i].Endpoints) > len(sl[j].Endpoints)
 }
+
+// returns a map of address types used by a service
+func getAddressTypesForService(service *corev1.Service) map[discovery.AddressType]struct{} {
+	serviceSupportedAddresses := make(map[discovery.AddressType]struct{})
+	// TODO: (khenidak) when address types are removed in favor of
+	// v1.IPFamily this will need to be removed, and work directly with
+	// v1.IPFamily types
+
+	// IMPORTANT: we assume that IP of (discovery.AddressType enum) is never in use
```

Comment:
I don't understand this comment - can you clarify?

Reply:
discovery.AddressType enum has a type IP which is deprecated and must not be used any more. @robscott can add more color to this.

Reply:
I see, I didn't realize it was deprecated